src/Dialog.js

```diff
@@ -88,30 +62,64 @@ function DialogHeader({theme, children, ...rest}) {
   )
 }
 
-function Dialog({children, ...props}) {
-  return (
+const Overlay = styled.span`
+  &:before {
+    position: fixed;
+    top: 0;
+    right: 0;
+    bottom: 0;
+    left: 0;
+    z-index: 80;
+    display: block;
+    cursor: default;
+    content: ' ';
+    background: transparent;
+    z-index: 99;
+    background: rgba(27, 31, 35, 0.5);
+  }
+`
+
+const Dialog = forwardRef(({children, onDismiss, isOpen, initialFocusRef, returnFocusRef, ...props}, forwardedRef) => {
+  const backupRef = useRef(null)
+  const modalRef = forwardedRef ?? backupRef
+  const closeButtonRef = useRef(null)
+
+  const onCloseClick = () => {
+    onDismiss()
+    if (returnFocusRef && returnFocusRef.current) {
+      returnFocusRef.current.focus()
+    }
+  }
+
+  const {getDialogProps} = useDialog({modalRef, onDismiss, isOpen, initialFocusRef, closeButtonRef, returnFocusRef})
```

Comment:
I wonder if instead of making `returnFocusRef` a parameter here, if we should just wrap the consumer's `onDismiss` function in a new function that is essentially the current contents of `onCloseClick` and then pass that as the `onDismiss` to `useDialog`.

Just noticed the duplication between the close button' `onClick`, `onClickOutside` in `useDialog` and the `ESCAPE` key handler in `useDialog` and trying to see if there is a way to DRY it up a little bit.